infrastructure/src/test/java/org/corfudb/infrastructure/CorfuSequencerServerTest.java

```diff
@@ -0,0 +1,528 @@
+import static org.corfudb.runtime.proto.RpcCommon.SequencerMetricsMsg.SequencerStatus;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@Slf4j
+// TODO (Chetan): Remove this after completion
+@Ignore("Still under development.")
+public class CorfuSequencerServerTest {
+
+    private final AtomicInteger requestCounter = new AtomicInteger();
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
+    @Mock
+    Map<UUID, StreamAddressSpace> mockStreamAddressSpaceMap;
+    // The SequencerServer instance used for testing
+    private SequencerServer sequencerServer;
+    // Objects that need to be mocked
+    @Mock
+    private ServerContext mockServerContext;
+    @Mock
+    private IServerRouter mockServerRouter;
+    @Mock
+    private ChannelHandlerContext mockChannelHandlerContext;
+    @Spy
+    private SequencerServer.FactoryHelper spyFactoryHelper;
+
+    /**
+     * A helper method that creates a basic message header populated
+     * with default values.
+     *
+     * @param ignoreClusterId indicates if the message is clusterId aware
+     * @param ignoreEpoch     indicates if the message is epoch aware
+     * @return the corresponding HeaderMsg
+     */
+    private HeaderMsg getBasicHeader(boolean ignoreClusterId, boolean ignoreEpoch) {
+        return getHeaderMsg(requestCounter.incrementAndGet(), PriorityLevel.NORMAL, 0L,
+                getUuidMsg(DEFAULT_UUID), getUuidMsg(DEFAULT_UUID), ignoreClusterId, ignoreEpoch);
+    }
+
+    /**
+     * A helper method that compares the base fields of two message headers.
+     * These include the request ID, the epoch, the client ID, and the cluster ID.
+     *
+     * @param requestHeader  the header from the request message
+     * @param responseHeader the header from the response message
+     * @return true if the two headers have the same base field values
+     */
+    private boolean compareBaseHeaderFields(HeaderMsg requestHeader, HeaderMsg responseHeader) {
+        return requestHeader.getRequestId() == responseHeader.getRequestId() &&
+                requestHeader.getEpoch() == responseHeader.getEpoch() &&
+                requestHeader.getClientId().equals(responseHeader.getClientId()) &&
+                requestHeader.getClusterId().equals(responseHeader.getClusterId());
+    }
+
+    /**
+     * Perform the required preparation before running individual tests.
+     * This includes preparing the mocks and initializing the DirectExecutorService.
+     */
+    @Before
+    public void setup() {
```

Comment:
I notice that you have some repeated initialization code, such as the initialization of the server, in a handful of tests. Would it be simpler to move such snippets here? This would also help you break the tests down into smaller units without having to repeat the initialization work.

Reply:
Thanks for pointing it out. I was still in the middle of some changes. I will keep in mind to move the common code here.